av1an-core/src/grain.rs

```diff
@@ -28,10 +30,51 @@ const PQ_C1: f32 = 3424. / 4096.;
 const PQ_C2: f32 = 32. * 2413. / 4096.;
 const PQ_C3: f32 = 32. * 2392. / 4096.;
 
+const BT1886_WHITEPOINT: f32 = 203.;
+const BT1886_BLACKPOINT: f32 = 0.1;
+const BT1886_GAMMA: f32 = 2.4;
+
+// BT.1886 formula from https://en.wikipedia.org/wiki/ITU-R_BT.1886.
+//
+// TODO: the inverses, alpha, and beta should all be constants
+// once floats in const fns are stabilized and `powf` is const.
+// Until then, `inline(always)` gets us close enough.
+
+#[inline(always)]
+fn bt1886_inv_whitepoint() -> f32 {
+  BT1886_WHITEPOINT.powf(1.0 / BT1886_GAMMA)
+}
+
+#[inline(always)]
+fn bt1886_inv_blackpoint() -> f32 {
+  BT1886_BLACKPOINT.powf(1.0 / BT1886_GAMMA)
+}
+
+/// The variable for user gain:
+/// `α = (Lw^(1/λ) - Lb^(1/λ)) ^ λ`
+#[inline(always)]
+fn bt1886_alpha() -> f32 {
+  (bt1886_inv_whitepoint() - bt1886_inv_blackpoint()).powf(BT1886_GAMMA)
+}
+
+/// The variable for user black level lift:
+/// `β = Lb^(1/λ) / (Lw^(1/λ) - Lb^(1/λ))`
+#[inline(always)]
+fn bt1886_beta() -> f32 {
+  bt1886_inv_blackpoint() / (bt1886_inv_whitepoint() - bt1886_inv_blackpoint())
+}
+
 impl TransferFunction {
   pub fn to_linear(self, x: f32) -> f32 {
     match self {
-      TransferFunction::BT1886 => x.powf(2.8),
+      TransferFunction::BT1886 => {
+        // The screen luminance in cd/m^2:
+        // L = α * max((x + β, 0))^λ
+        let luma = bt1886_alpha() * 0f32.max(x + bt1886_beta()).powf(BT1886_GAMMA);
```

Comment:
`max(x + β, 0)` as opposed to just `x + β` inside this function appears to have the same value for all inputs, see https://www.desmos.com/calculator/1lxvcqwtny

Reply:
Nevermind, I completely messed up that comparison.

Reply:
I.e. we can drop the max and this will return the same results? sounds good to me. Interesting that the formula had the `max` there considering all inputs are restricted to 0 <= x <= 1.

Reply:
I guess we could still remove it then? I also noticed that the inverse of `to_linear` seemed to be derived from a version of `to_linear` without the `max`, so I guess it would make sense if we could remove it.